backend/src/lib.rs

```diff
@@ -192,8 +192,7 @@ impl LTN {
             .map(|feature| {
                 let mut polygon = Polygon::try_from(feature).map_err(err_to_js)?;
                 self.map.mercator.to_mercator_in_place(&mut polygon);
-                make_polygon_valid(&mut polygon);
-                Ok(polygon)
+                make_polygon_valid(&polygon).map_err(err_to_js)
```

Comment:
Big change here! Previously we were calling `make_polygon_valid` but throwing away the result; it doesn't modify anything in-place. The only effect was to possibly crash in debug mode. I haven't noticed any behavioral differences now that this is using the result. Was the prior behavior deliberate?

Reply:
Given the confusing method name and the mutable references, it's almost surely an error on my part. 

I think the intention was, in the case of a Multipolygon, since we don't handle MultiPolygons here, to continue on with one of its inner Polygons.

We'll no longer do that at the other `make_polygon_valid` callsite. Maybe that's fine and not something I should have been trying to avoid.